### ISSUE TITLE
Make crossbow and bow crosstrain.

### ISSUE THREAD

**Issue by G0rg0** [jeremygurr/dcssca#384]
It just doesn't make sense that they don't.

**Comment by jeremygurr** [comment]
Yeah I guess that makes as much sense as anything else crosstraining.

**Comment by yrmvgh** [comment]
How about staves and unarmed? Very Friar Tuck. Very shao lin. Everyone happy.

**Comment by G0rg0** [comment]
I dunno, UC seems very unique to me, I can't see it logically crosstraining anything.

**Comment by Sandman25DCSS** [comment]
How about removing all crosstraining except short blade-long blades? Short blades become bad late game but everything else has good weapons.

**Comment by jeremygurr** [comment]
I'm on the fence here. 

There is value in removing most crosstraining, as it makes those earlier game decisions more important and the distinction between weapons greater. But this assumes that there are adequate end game weapons in each category. I think this should even be true for short swords. They should have their own end game weapons that have strategic trade offs. Maybe many melee fighters would train two weapon skills to have that extra option. But that decision is a lot more meaningful without crosstraining. And that also makes Ash's skill transfer power a much bigger deal, might even be worth enduring her wrath to get that late game skill change in some cases. 

But then we have the other side. Having many crosstraining options favors hybrid and wanderer styles, where you make the best of what you come across. This is especially nice for new players who don't fully understand the implications of various weapons. Too bad there isn't a way to get both advantages. Or is there...?

Option 1: standard mode has many crosstraining options, for weapons and magic schools. Nightmare mode has no crosstraining at all. Challenge mode is in between. This is simple to understand and somewhat consistent with the needs of new players vs experienced onces. This has the advantage of another interesting difficulty differentiator. I don't want too many differences, but this is a significant one that could replace some of the less globally impactful ones. 

Option 2: crosstraining is species specific. We already have humans able to crosstrain things that other species can't. I think this makes the humans have a different and consistent play style. We could expand this to a few other species, and make some species have no crosstraining. This has the advantage of being another species differentiator, and being able to play both focussed and flexible styles in any difficulty mode. 

Option 3: get rid of all crosstraining and introduce a new kind of manual that adds cross training options. These could be a little more common than pure skill boosting manuals, since they aren't nearly as valuable, but each would still introduce one crosstraining group, and they would still be rare enough not to obliterate the benefits of removing crosstraining. A few of these crosstraining manuals could also be included as starting equipment for a wanderer, for example. If we went this route, I'd like to keep humans able to crosstrain as part of their species uniqueness.

**Comment by Sandman25DCSS** [comment]
Option 1 is the easiest to implement and most straightforward I think. Hybrids are not that interested in crosstraining, they usually stick to one weapon category because they need all the XP they can get. It's berserkers who can easily switch from one weapon category to another because of god gift and no other skills to train.

**Comment by G0rg0** [comment]
I never really saw crosstraining as a problem in need of solving. It's not OP really, it just opens up some options and just makes sense in most cases. 

If you are superb with an axe, it would be silly to assume that handed a polearm, you'd have literally no carryover in skill, and be just as bad as someone who has only used bows or swords.

**Comment by jeremygurr** [comment]
I agree that it is more intuitive that skills would crosstrain, and intuitiveness is an important design objective. But differentiation which leads to increased choices is pretty important too as far as design objectives, and that is probably better achieved by less crosstraining. 

I agree that option 1 is the easiest. But I'm guessing Gorgo would like to be able to play nightmare and still be able to crosstrain. Although I won't always be able to please everyone, I do try to find an everyone-wins solution when it isn't too difficult. 

We could just have an rc file option for it, but I'm against options that dramatically change game difficulty because that makes tournaments and leaderboards meaningless. Options are more for changing interface and non-game changing features. Although options _are_ good for temporary features we are experimenting with. 

Option 3 is still fairly easy to implement and provides somewhat of a compromise, but maybe not enough for what some would like to see. 

What do you think Gorgo? Would you prefer option 1 or 3, or something else? I'm guessing Sandman would be fine with 3 also?

**Comment by G0rg0** [comment]
Well, I don't see how removing crosstraining _increases_ choices, but if you guys see value in it, I'd say give option 1 a go and see how it plays out.

**Comment by Sandman25DCSS** [comment]
With option 3 player needs to find _both_ the "manual" and a superb weapon which is better than everything they have in already trained weapon category, that won't happen often I think.

**Comment by Sandman25DCSS** [comment]
G0rg0
Crosstraining often decreases choices because it makes switching too easy and thus obviously optimal. Just today I was playing a DsAs (http://crawl.homedns.org/morgue/Sandman25/morgue-Sandman25-20160429-233523.txt) and I found scimitar when my Short Blades was 4.7 and Long Blades was 1.1 (from crosstraining). Of course I instantly stopped training short blades and started training long blades because crosstraining alone would be enough to bring dagger to min delay if I lived long enough

**Comment by jeremygurr** [comment]
So I guess I will do option one, and take crosstraining from humans, but I do like the new manual idea. I think that adds some fun flavor and twists to challenge and nightmare modes in a similar way that normal manuals do.

**Comment by G0rg0** [comment]
Sandman: 
I'll preface with saying I am fine with the proposed change, but I read your example as having increased your choices by letting you move to longblades more easily.

**Comment by yrmvgh** [comment]
I always thought of cross training as an important fallback for situations where you train your heart out on one weapon type (let's say axes, because it's usually easy to find a mace in my experienc), and just can't find a decent endgame-quality option. Even acquirement and brand scrolls (if you find them) fail you. So you fallback to crosstraining and grab a mace - it still costs you a bunch of skill points, but the sting of being screwed by RNG isn't quite so bad.

It's not like when you find Englaciate you're stuck with a crappy englaciate because it's a mediocre randart and thus you can't brand/enchant it. It just works!
